### ISSUE TITLE
Model size regularization and loss performance - GPU model worse than CPU model

### ISSUE THREAD

**Issue by bduclaux** [catboost/catboost#1392]
Hello,

I am training a regression model with approx 4m records and 13 features, with 6 categorical ones with high cardinality for a few of them. Loss is RMSE. Records have a weight column. Catboost version is 0.24 on Linux (cmdline version).

When training on CPU, the model size is about 800Mbytes, and the loss is approx 0.017 after 500 iterations. Each iteration improves the loss.

When training on GPU, the model size explodes to approx 5GB, and the loss is significantly worse, around 0.025. The use of model-size-reg reduces the model size, but no matter the setting, the loss is always worse than on CPU.
I have tried to set a manual learning rate on GPU with the CPU autodetect value, but still the GPU model is way behind the CPU one in terms of loss.

```
Cmdline for CPU training:
catboost fit --task-type CPU --learn-set train.txt --cd columns.cd --loss-function RMSE -i 500 --max-ctr-complexity 2

Cmdline for GPU training:
catboost fit -w 0.289 --task-type GPU --devices 0-3  --learn-set train.txt --cd columns.cd --loss-function RMSE -i 500 --max-ctr-complexity 2 --model-size-reg 0.5
```

I suspect there is an issue in the GPU implementation for this particular dataset, as two categorical features have more than 100k and 500k values each.

Dataset and columns descriptors made available to the Yandex team.

**Comment by botasb** [comment]
Hi. 
Did you confirm this issue? I am experiencing similar mismatches between GPU and CPU regression issues.

**Comment by bduclaux** [comment]
Hi

The problem is linked to different default settings for model size regularization between GPU and CPU.
See https://catboost.ai/docs/references/model-size-reg.html
You might want to adjust the GPU settings to you particular problem.

**Comment by LyzhinIvan** [comment]
The main issue about bad performance on GPU is #1408. I close this as duplicate.
